extension/src/components/ConfirmLoanWizard.tsx

```diff
@@ -19,99 +38,148 @@ const ConfirmLoanWizard = ({ onCancel, onSuccess, loanToSign }: ConfirmLoanWizar
     });
     let [signedTransaction, setSignedTransaction] = useState("");
 
-    const onSigned = (tx: string) => {
-        setSignedTransaction(tx);
-        nextStep();
-    };
+    let { isPending: isSigning, run: sign } = useAsync({
+        deferFn: async () => {
+            let signedTransaction = await signLoan();
+            setSignedTransaction(signedTransaction);
+            nextStep();
+        },
+    });
 
-    const downloadLoanBackup = async () => {
-        const loanBackup = await createLoanBackup(signedTransaction);
-        const file = new Blob([JSON.stringify(loanBackup)], { type: "text/json" });
-        const url = URL.createObjectURL(file);
-        // Note: it would be nicer to open a dialog to download the file. However, we would lose focus
-        // of the window. We don't want that, hence we just open a new tab with the content
-        window.open(url, "_blank");
-        nextStep();
-    };
+    const { isPending: isDownloading, run: downloadLoanBackup } = useAsync({
+        deferFn: async () => {
+            const loanBackup = await createLoanBackup(signedTransaction);
+            const file = new Blob([JSON.stringify(loanBackup)], { type: "text/json" });
+            const url = URL.createObjectURL(file);
+            // Note: it would be nicer to open a dialog to download the file. However, we would lose focus
+            // of the window. We don't want that, hence we just open a new tab with the content
+            window.open(url, "_blank");
+            nextStep();
+        },
+    });
 
-    const onPublish = async () => {
-        await confirmLoan(signedTransaction);
-        onSuccess();
-    };
+    const { isPending: isPublishing, run: publish } = useAsync({
+        deferFn: async () => {
+            await confirmLoan(signedTransaction);
+            onSuccess();
+        },
+    });
 
     return (
         <VStack width="100%">
             <Steps activeStep={activeStep}>
                 <Step label={"Sign"} key={"sign"} icon={FiClipboard}>
                     <Flex py={4}>
-                        <ConfirmLoan
-                            loanToSign={loanToSign}
-                            onCancel={onCancel}
-                            onSigned={onSigned}
-                        />
+                        <form
+                            onSubmit={async e => {
```

Comment:
Doesn't need to be async. I think there is a typescript lint for such cases!

Reply:
There are quite a lint errors. I created a ticket for that: https://github.com/comit-network/waves/issues/236